File: C++/Row with max 1s.cpp
class Solution {
  public:
    int rowWithMax1s(vector<vector<int> > &arr) {
        int len=0;
        int ans=-1;
        int m=arr.size();
        int n=arr[0].size();
        int j=n-1;
        for(int i=0;i<m;i++){
           while(j>=0&&arr[i][j]==1){
               j--;
           }
           if(n-j-1>len){
               ans=i;
               len=n-j-1;
           }
        }
        return ans;
    }
};

File: JAVA/Row with max 1s.java
class Solution {
    public int rowWithMax1s(int arr[][]) {
        int len=-1;
        int idx=0;
        for(int i=0;i<arr.length;i++){
            int count=0;
            for(int j=0;j<arr[0].length;j++){
                if(arr[i][j]==1)count++;
            }
            if(count!=0){
               if(count>len)idx=i; 
            len=Math.max(count,len);
            }
        }
        if(len==-1)return -1;
        return idx;
    }
} 
